lightning/backend/local.go

```diff
@@ -851,6 +863,9 @@ func (local *local) writeAndIngestByRange(
 	defer iter.Close()
 	// Needs seek to first because NewIter returns an iterator that is unpositioned
 	hasKey := iter.First()
+	if iter.Error() != nil {
+		return errors.Annotate(iter.Error(), "failed to read the first key")
+	}
```

Comment:
What about iter.Last() at line 877?

Reply:
What about this comment? @glorv

Reply:
fixed

Reply:
do we need to check `Valid()` after `iter.Last()`, as we did for `iter.First()`?

Reply:
yes,  see #880-881 @lance6716

Reply:
Do you mean `iter.Error()` here could cover `iter.Valid()`? For `iter.First()` why we need them both (maybe iterator will keep valid after normal operation on iterator 🤔 )

Reply:
we only need to check iter.Error() != nil which  indicates that something goes wrong. iter.Error() != nil always means iter.Valid() is false. if iter.Error() == nil and iter.Valid() is false, this means the iterator iters to the end, this is ok to our logic.